client/db/src/lib.rs

```diff
@@ -594,6 +594,37 @@ impl<Block: BlockT> sc_client_api::blockchain::Backend<Block> for BlockchainDb<B
 	fn has_indexed_transaction(&self, hash: &Block::Hash) -> ClientResult<bool> {
 		Ok(self.db.contains(columns::TRANSACTION, hash.as_ref()))
 	}
+
+	fn block_indexed_body(&self, id: BlockId<Block>) -> ClientResult<Option<Vec<Vec<u8>>>> {
+		match self.transaction_storage {
+			TransactionStorageMode::BlockBody => Ok(None),
```

Comment:
There's no default notion for indexed data. `BlockBody` means that no indexed data is stored, so `None` looks appropriate. Also it would be incorrect to return `Some` for blocks that are truly unknown.